File: Java-CodeServer/src/main/resources/js/prompto/runtime/ErrorVariable.js
var Variable = require("./Variable").Variable;
var EnumeratedCategoryType = null;

exports.resolve = function() {
    EnumeratedCategoryType = require("../type/EnumeratedCategoryType").EnumeratedCategoryType;
};

function ErrorVariable(name) {
	Variable.call(this, name, new EnumeratedCategoryType("Error"));
	return this;
}

ErrorVariable.prototype = Object.create(Variable.prototype);
ErrorVariable.prototype.constructor = ErrorVariable;

ErrorVariable.prototype.toString = function() {
	return this.name;
};

ErrorVariable.prototype.getType = function(context) {
	return new EnumeratedCategoryType("Error");
};

exports.ErrorVariable = ErrorVariable;

File: Java-Core/src/main/java/prompto/runtime/ErrorVariable.java
package prompto.runtime;

import prompto.error.SyntaxError;
import prompto.grammar.Identifier;
import prompto.type.EnumeratedCategoryType;
import prompto.type.IType;

public class ErrorVariable extends Variable {

	public ErrorVariable(Identifier name) {
		super(name,null);
	}
	
	@Override
	public String toString() {
		return getId().toString();
	}
	
	@Override
	public IType getType(Context context) throws SyntaxError {
		return new EnumeratedCategoryType(new Identifier("Error"));
	}
	
}
